### ISSUE TITLE
A path cleaning utility for removing the site- from URLs.

### ISSUE THREAD

**Issue by alan-cole** [dpc-sdp/ripple-framework#1049]
### Describe the feature

I propose we add a utility that can provide an interface to pass a url in as a prop and have it returned with the `/site-X/` removed.

### Motivation

The same pattern of stripping the `/site-X/` is used in 5 different components, and is likely to be copy pasted for any other uses.

![Screenshot 2024-02-23 at 1 54 59 pm](https://github.com/dpc-sdp/ripple-framework/assets/12739575/008e4a10-900a-44fc-9a22-0bc1ee433c7c)

### Proposal

In `ripple-tide-api/src/utils/mapping-utils.ts`, adding:

```ts
export const getCleanURL = (url, replace = '') => {
  return url.replace(/\/site-(\d+)/, replace)
}
```

Alternatively, exporting the regular expression `\/site-(\d+)/` may be valuable:

```
export const SITE_MATCH = /\/site-(\d+)/
```

### Benefits

Less copy / paste of common patterns.

### Drawbacks

More dependencies between components?

### Adoption strategy

Drive adoption through documentation.

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be created outside of Ripple-framework as a layer?

### Final checks

- [X] I have read the [contribution guide](https://ripple.sdp.vic.gov.au/design-system/about/contributing) for information on submitting a feature request.
- [X] I have checked the [discussions](https://github.com/dpc-sdp/ripple-framework/discussions) and [issues](https://github.com/dpc-sdp/ripple-framework/issues) for duplicate feature requests.

**Comment by dylankelly** [comment]
makes sense @alan-cole, something you would like to contribute? PRs welcome.

**Comment by alan-cole** [comment]
@dylankelly - I've raised a PR for feedback.

Some slight changes to my initial proposal:

1. I've added it to `ripple-tide-api`, as I feel the `site-x` is an implementation detail that comes from tide, so a utility to strip it would relate to a tide (similar to the normalize image url utility in `ripple-tide-api`).
2. Changed the name from `getCleanURL` to `stripSiteId` to make it clearer what the function does, though I'm happy to name it whatever. Alternatives could be `formatUrl` or `cleanUrl` etc.

**Comment by dylankelly** [comment]
Thanks @alan-cole, I've approved the PR and have assigned to the Ripple 2.8 release milestone.

**Comment by dylankelly** [comment]
resolved in #1084
